### PR TITLE
Deal.get_all.

### PR DESCRIPTION
**Pull request by westonplatter**

### CODE REVIEW

**Comment by hajder** [comment]
Hi @westonplatter, thanks for our first pull request! Yay!

I will take a look at it over weekend when I have more time. We're planning on a major refactoring of this gem since we use it a lot at @stagelink, any input on that is very welcome.

**Comment by westonplatter** [comment]
@hajder sounds great. no rush, just wanted to share changes if they're valuable with you and the team.

File: lib/pipedrive/deal.rb [hunk 1]
```diff
@@ -12,8 +12,24 @@ def self.filter(filter_id, fetch_all_pages = true)
         break unless fetch_all_pages
         start += limit
       end
+
+      return acc
+    end
+
+    def self.get_all(ops={})
```
Comment (hajder) [inline lib/pipedrive/deal.rb]: Would be great if you moved this to `Pipedrive::Base` so all the classes could benefit from this

Comment (hajder) [inline lib/pipedrive/deal.rb]: I think writing `Deal.all` would be nicer and more in line with ActiveRecord semantics

File: lib/pipedrive/deal.rb [hunk 2]
```diff
@@ -12,8 +12,24 @@ def self.filter(filter_id, fetch_all_pages = true)
...
+      acc = []
+      start = ops[:start] || 0
+      limit = ops[:limit] || 100
+      fetch_all_pages = ops[:fetch_all_pages] || true
```
Comment (hajder) [inline lib/pipedrive/deal.rb]: Does is make sense to have this logic in here? I think it should just get all objects

**Pull request closed without merging**